src/Microsoft.DotNet.Interactive.CSharpProject/BuildCacheFileUtilities.cs

```diff
@@ -0,0 +1,146 @@
+            <ItemGroup>
+              <ProjectData Include=""ProjectGuid=$(ProjectGuid)"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""%(ProjectReference.Identity)"">
+                <Prefix>ProjectReferences=</Prefix>
+                <Type>Array</Type>
+              </ProjectData>
+              <ProjectData Include=""ProjectFilePath=$(MSBuildProjectFullPath)"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""LanguageName=C#"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""PropertyTargetPath=$(TargetPath)"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""%(Compile.Identity)"" Condition=""!$([System.String]::new('%(Compile.Identity)').Contains('obj\'))"">
+                <Prefix>SourceFiles=</Prefix>
+                <Type>Array</Type>
+              </ProjectData>
+              <ProjectData Include=""%(ReferencePath.Identity)"">
+                <Prefix>References=</Prefix>
+                <Type>Array</Type>
+              </ProjectData>
+              <ProjectData Include=""%(Analyzer.Identity)"">
+                <Prefix>AnalyzerReferences=</Prefix>
+                <Type>Array</Type>
+              </ProjectData>
+              <ProjectData Include=""$(DefineConstants)"">
+                <Prefix>PreprocessorSymbols=</Prefix>
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""PropertyLangVersion=$(LangVersion)"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""PropertyOutputType=$(OutputType)"">
+                <Type>String</Type>
+              </ProjectData>
+            </ItemGroup>
+        
+            <!-- Split PreprocessorSymbols into individual items -->
+            <ItemGroup>
+              <PreprocessorSymbolItems Include=""$(DefineConstants.Split(';'))"" />
+            </ItemGroup>
+        
+            <!-- Transform the ProjectData and PreprocessorSymbolItems to include the prefix -->
+            <ItemGroup>
+              <ProjectDataLines Include=""$([System.String]::Format('{{0}}{{1}}', %(ProjectData.Prefix), %(ProjectData.Identity)))"" />
+              <ProjectDataLines Include=""$([System.String]::Format('PreprocessorSymbols={{0}}', %(PreprocessorSymbolItems.Identity)))"" />
+            </ItemGroup>
+        
+            <!-- Write collected project data to a file -->
+            <WriteLinesToFile Lines=""@(ProjectDataLines)""
+                              File=""$(MSBuildProjectFullPath).interactive.workspaceData.cache""
+                              Overwrite=""True""
+                              WriteOnlyWhenDifferent=""True"" />
+          </Target>
+        </Project>";
+
+    public record BuildProjectData
```

Comment:
Why `record`?

Reply:
changed to class.

I was using record because this is read-only data